server/parser.go

```diff
@@ -0,0 +1,65 @@
+package server
+
+import (
+	"fmt"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	vitess "github.com/dolthub/vitess/go/vt/sqlparser"
+
+	"github.com/dolthub/doltgresql/postgres/parser/parser"
+	"github.com/dolthub/doltgresql/server/ast"
+)
+
+var _ sql.Parser = &PostgresParser{}
+
+type PostgresParser struct{}
+
+func NewPostgresParser() *PostgresParser { return &PostgresParser{} }
+
+func (p *PostgresParser) ParseSimple(query string) (vitess.Statement, error) {
+	stmt, _, _, err := p.ParseWithOptions(query, ';', false, vitess.ParserOptions{})
+	return stmt, err
+}
+
+func (p *PostgresParser) Parse(_ *sql.Context, query string, multi bool) (vitess.Statement, string, string, error) {
+	return p.ParseWithOptions(query, ';', multi, vitess.ParserOptions{})
+}
+
+func (p *PostgresParser) ParseWithOptions(query string, delimiter rune, _ bool, _ vitess.ParserOptions) (vitess.Statement, string, string, error) {
+	q := sql.RemoveSpaceAndDelimiter(query, delimiter)
+	if vitessStmt, err := vitess.Parse(q); err == nil {
```

Comment:
Why do we use the `vitess` parser here? Shouldn't we **always** use our parser?

Reply:
Ah I temporarily added it to be able to run the tests because we currently don't support `USE` statement.

Reply:
I just added `TODO` for it and it will only use `vitess` parser for `USE` statements for now